keps/sig-api-machinery/3903-unknown-version-interoperability-proxy/README.md

```diff
@@ -254,12 +254,36 @@ the number if needed, but upgrades aren't that frequent and few resources are
 changed on releases, so these requests should not be common. We will count them
 with a metric.
 
+There could be a large volume of requests for a specific resource which might result in the identified apiserver being unable to serve the proxied requests. This scenario should not occur too frequently, since resource types which have large request volume should not be added or removed during an upgrade -- that would cause other problems, too.
+
 TODO: security / cert stuff.
 
 ## Design Details
 
-TODO: explanation of how the handler will determine a request is for a resource
-that should be proxied.
+Update the handler stack to detect the condition where proxying of the request to an alternate apiserver is needed.
+```
+func (h *handler) ServeHttp(req *http.Request){
+
+  // find the apiserverid that knows of the request's GVR
+  APIServerIdForResource = StorageVersion.getServerIdsForGVR(req))
+  
+  if (!APIServerIdsForResource) {
+    // no apiserver found that knows about req.resource
```

Comment:
this means the resource is an aggregated api or a CR -- in both cases this filter should do nothing, just pass the request through the filter chain.

Reply:
Okay, but how and where is it ultimately served then?

Reply:
Specifically, the aggregator proxies to aggregated apis in its main handler (which happens after all the filters); if the request is not for an aggregated api, it goes to the delegate, the built-in apiserver; if the request is not for a built-in, it goes to the next delegate, the extensions-apiserver (CRDs); if the request is not for a CRD either, it goes to the final delegate, which is just a 404 handler. But in this case a 404 is correct, since the pseudocode you've added essentially verifies the request is not for a built-in that *any* apiserver knows about.

Reply:
Got it, thanks for the detailed explanation. Is it possible when we query the StorageVersion that we find a resource was registered but somehow no apiserver was found that reported it? Perhaps no..

Reply:
Yes, if the apiserver that served it shut down and the records haven't been updated yet. We can limit this window to 5s (IIRC) by looking at the lease object for the apiserver we pick to verify it hasn't expired. In that case -- there's only one apiserver choice and it doesn't have a fresh lease -- we can serve 503, I think.

Reply:
Makes sense!